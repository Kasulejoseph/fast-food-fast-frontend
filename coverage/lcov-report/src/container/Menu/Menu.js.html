<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/container/Menu/Menu.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/container/Menu</a> Menu.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/9</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import {connect} from 'react-redux'
import {myMenu} from '../../actions/fetchAction'
import PostForm from '../../components/PostForm/PostForm'
class Postmenu extends React.Component {
<span class="fstat-no" title="function not covered" >    co</span>nstructor(props) {
<span class="cstat-no" title="statement not covered" >        super(props);</span>
<span class="cstat-no" title="statement not covered" >        this.state = {</span>
            meal: '',
            description: '',
            price: '',
            image: ''
        };
<span class="cstat-no" title="statement not covered" >        this.onChange = this.onChange.bind(this)</span>
<span class="cstat-no" title="statement not covered" >        this.onSubmit = this.onSubmit.bind(this)</span>
    }
<span class="fstat-no" title="function not covered" >    on</span>Change(e){
<span class="cstat-no" title="statement not covered" >        this.setState({[e.target.name]: e.target.value});</span>
    }
<span class="fstat-no" title="function not covered" >    on</span>Submit(e){
<span class="cstat-no" title="statement not covered" >        e.preventDefault();</span>
        const menu = <span class="cstat-no" title="statement not covered" >{</span>
            meal: this.state.meal,
            description: this.state.description,
            price: Number(this.state.price),
            image: 'image.jpg'
        };
<span class="cstat-no" title="statement not covered" >        this.props.myMenu(menu)</span>
    }
<span class="fstat-no" title="function not covered" >    re</span>nder() {
<span class="cstat-no" title="statement not covered" >        return (</span>
          &lt;PostForm onChange={this.onChange} onSubmit={this.onSubmit}/&gt;        
        );
    }
}
&nbsp;
export default connect (
    null,
    {myMenu}
) (Postmenu)
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Jan 28 2019 15:44:35 GMT+0300 (East Africa Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
