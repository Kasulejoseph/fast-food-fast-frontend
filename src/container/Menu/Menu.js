import React from 'react';
import {connect} from 'react-redux'
import {myMenu} from '../../actions/feactAction'
import PostForm from '../../components/PostForm/PostForm'
import FixedNavbarExample from '../../container/NavBar/NavBar';
import Navbar from '../../components/NavBar/Navbar';


export class Postmenu extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            meal: '',
            description: '',
            price: '',
            image: '',
            collapse: false,
        };
        this.onChange = this.onChange.bind(this)
        this.onSubmit = this.onSubmit.bind(this)
        this.onClick = this.onClick.bind(this);

    }
    onChange(e){
        this.setState({[e.target.name]: e.target.value});
    }
    onClick() {
        this.setState({
            collapse: !this.state.collapse,
          });
      }
    onSubmit(e){
        e.preventDefault();
        const menu = {
            meal: this.state.meal,
            description: this.state.description,
            price: Number(this.state.price),
            image: 'image.jpg'
        };
        myMenu(menu)
    }
    render() {
        return (
            <>
            <Navbar onClick={this.onClick} isOpen ={this.state.collapse}/>,
          <PostForm onChange={this.onChange} onSubmit={this.onSubmit}/>  
          </>      
        );
    }
}

export default connect (
    null,
    {myMenu}
) (Postmenu)
