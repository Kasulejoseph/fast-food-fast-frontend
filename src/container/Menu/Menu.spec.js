import React from 'react';
import { shallow } from 'enzyme';
import { Postmenu } from './Menu';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

let props = {
  onChange: jest.fn(),
  onSubmit: jest.fn()
};
describe('Test Postmenu', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Postmenu {...props} />);
  });
  it('Postmenu snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('test handle onsubmit event', () => {
    wrapper.instance().onSubmit({ preventDefault() {} });
    expect(wrapper.state()).toEqual({
      active: "false",
      collapse: false,
      description: '',
      image: '',
      meal: '',
      price: ''
    });
  });

  it('should handle onchange function', () => {
    wrapper
      .instance()
      .onChange({ target: { name: 'email', value: 'kasule@andela.com' } });
    expect(wrapper.state('email')).toEqual('kasule@andela.com');
  });
});
