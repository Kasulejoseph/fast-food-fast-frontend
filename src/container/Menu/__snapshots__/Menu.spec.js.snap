// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Postmenu Postmenu snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Postmenu
    onChange={[MockFunction]}
    onSubmit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Navbar
          active="false"
          isOpen={false}
          onClick={[Function]}
        />,
        ",",
        <PostForm
          onChange={[Function]}
          onSubmit={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": "false",
          "isOpen": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      ",",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onChange": [Function],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Navbar
            active="false"
            isOpen={false}
            onClick={[Function]}
          />,
          ",",
          <PostForm
            onChange={[Function]}
            onSubmit={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": "false",
            "isOpen": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        ",",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onChange": [Function],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
